### ISSUE TITLE
dsv4 hc_pre.py: same disproven dep_gen rationale as decode_fwd (fixed in #797) — verify + align to --enable-dep-gen

### ISSUE THREAD

**Issue by ChaoWao** [hw-native-sys/pypto-lib#798]
## Summary

`models/deepseek/v4/hc_pre.py` carries a `# ci: no-dep-gen` marker and a
`--no-dep-gen` flag whose rationale claims the full-occupancy `pl.system.syncall`
is incompatible with `dep_gen`:

```
# ci: no-dep-gen  # CI marker: full-occupancy pl.system.syncall -> dep_gen (DFX) trips 507018 (pypto#1931)
```

This is the **origin** of the same rationale that was later copied into
`models/qwen3/14b/decode_fwd.py`, where it has now been **disproven** and fixed
in #797.

## Provenance

- Marker + `--no-dep-gen` introduced in **`bbd7d93` (#684)** — *"perf(dsv4/hc_pre):
  fuse into one syncall task; fold the gate phase into Phase D"* (2026-07-04).
- Copied into `decode_fwd.py` in **`e41c03e` (#756)** (2026-07-13).

## Why the rationale is suspect

`dep_gen` is an **AICPU-side** DFX collector for the producer→consumer task
graph; it does **not** occupy AICore cores, so it cannot starve a
full-occupancy `SyncAll` barrier. On the qwen3 side this was verified on a2a3
(device 2): a 2-layer `--enable-dep-gen` run of the RoPE-fused attention
(full-occupancy `SyncAll`) completes cleanly — deps.json written for both
programs, forward validates (argmax 16/16, logits 100% within 5e-2), no 507018,
no `init_dep_gen`/`halHostRegister` failure. See #797.

## What this issue asks

`hc_pre.py` is a **different** kernel (dsv4 Phase-D fused syncall), so the qwen3
result should **not** be extrapolated blindly. Action items:

1. Run `hc_pre.py` on dsv4 with dep_gen enabled (small problem size) and confirm
   whether it completes and writes `dfx_outputs/deps.json`, or actually trips
   507018.
2. If it completes cleanly (expected): apply the same fix as #797 — replace the
   hardcoded off + false rationale with a single opt-in `--enable-dep-gen`
   (matching `prefill_attention_swa` / `decode_sparse_attn` /
   `prefill_attention_csa`), drop the redundant `--no-dep-gen`, and remove the
   vestigial `# ci: no-dep-gen` marker (no workflow parses it — only `no-sim` /
   `devices=N` are consumed). Note `hc_pre.py`'s current wiring is impl-coupled
   (`enable_dep_gen=(HC_PRE_IMPL == "separate")`), so the toggle needs care.
3. If it genuinely trips 507018: keep it off, but **correct the rationale** to
   describe the real mechanism (and confirm/soften the `pypto#1931` reference).

## Related

- #797 — the qwen3 `decode_fwd.py` fix (rationale disproven, toggle restored).
- #684 — origin of the marker.
- #756 — where it was copied into qwen3.

**Comment by zhangqi-chen** [comment]
Closing as done — verified against main `e058d747`.

The disproven `dep_gen` rationale is no longer in the tree:

- No `# ci: no-dep-gen` marker remains anywhere under `models/`.
- `models/deepseek_v4_pro/hc_pre.py:860` now documents `--no-dep-gen` as "Deprecated no-op retained for CLI compatibility; dep_gen follows --impl", i.e. the flag no longer suppresses dep_gen and the kernel is aligned with `--enable-dep-gen` like the rest of the tree. This landed in #1014.

Both halves of this issue's ask (verify + align) are therefore satisfied.
